src/TypeInference/Pattern.ml

```diff
@@ -330,6 +331,26 @@ and check_pattern_schemes ~env ~scope pats schs =
 
   | [], _ :: _ | _ :: _, [] -> assert false
 
+let infer_optional_arg_scheme env (arg : S.arg) =
+match arg with
+| ArgAnnot(pat, sch) ->
+  let sch_pos = sch.sch_pos in
+  let sch = Type.tr_scheme env sch in
+  assert (T.Scheme.is_monomorphic sch);
```

Comment:
There is no invariant that ensure this. User can write an invalid program that violates this assertion:
```
let foo {?x:{x:Int} -> Int} = x
```

The implementation should check if this scheme is monomorphic, and if not, display a readable error for the user.